src/Maui.RevenueCat.InAppBilling/Platforms/iOS/Extensions/RCOfferingsExtensions.cs

```diff
@@ -12,72 +12,27 @@ public static List<OfferingDto> ToOfferingDtoList(this RCOfferings offerings)
 
         foreach (var offer in offerings.All.Values)
         {
+            string? jsonString = null;
+
+            if (!offer.Metadata.IsNullOrEmpty())
+            {
+                var jsonData =
+                    NSJsonSerialization.Serialize(offer.Metadata, NSJsonWritingOptions.PrettyPrinted, out var error);
+                jsonString = error is null
+                    ? NSString.FromData(jsonData, NSStringEncoding.UTF8)
+                    : null;
+            }
+
             var offeringDto = new OfferingDto()
             {
                 Identifier = offer.Identifier,
                 AvailablePackages = offer.AvailablePackages.ToPackageDtoList(),
                 IsCurrent = offer.Identifier == offerings.Current?.Identifier,
-                Metadata = offer.Metadata.ToDictionary<KeyValuePair<NSString, NSObject>, string, object?>(item => item.Key,
-                    item => item.Value.ToDotNetObject())
+                Metadata = jsonString
             };
             offeringDtos.Add(offeringDto);
```

Comment:
Here we are mixing 2 concerns:
1) mapping native object to `OfferingDto`
2) a way how to parse iOS `NSDictionary` to json string

So I would propose to extract the serialization logic to separate method in `NsDictionaryExtensions`

Something like:
```
    internal static string? ToJson<T, U>(this NSDictionary<T, U> dictionary)
        where T : NSObject
        where U : NSObject
    {
        if (dictionary.IsNullOrEmpty())
        {
            return null;
        }

        var jsonData = NSJsonSerialization.Serialize(dictionary, NSJsonWritingOptions.PrettyPrinted, out var error);

        return error is null
            ? NSString.FromData(jsonData, NSStringEncoding.UTF8)
            : null;
    }
```

and then simply use:
```
Metadata = offer.Metadata.ToJson()
```

but after implementing this into iOS we are creating small inconsistency between Android and iOS code. Good practice is to keep them as similar as possible. So also please create `IDictonaryExtensions` in Android platform extensions folder where would be similar implementation

```
using Maui.RevenueCat.InAppBilling.Extensions;
using Org.Json;
using System.Collections;

namespace Maui.RevenueCat.InAppBilling.Platforms.Android.Extensions;

internal static class IDictonaryExtensions
{
    internal static string? ToJson<T, U>(this IDictionary<T, U> dictionary)
    {
        if (dictionary.IsNullOrEmpty())
        {
            return null;
        }

        return new JSONObject((IDictionary)dictionary).ToString().ToString();
    }
}
```

That way mapping in both implementations is identical and the only thing that is different are underlying type operations extracted into extension methods.